src/field/vector2d.cxx

```diff
@@ -148,6 +149,8 @@ Vector2D* Vector2D::timeDeriv() {
 /////////////////// ASSIGNMENT ////////////////////
 
 Vector2D & Vector2D::operator=(const Vector2D &rhs) {
+  fielddatamesh = rhs.fielddatamesh;
```

Comment:
I don't recall seeing similar statements for operator= in the fields, should there be?

Reply:
I think there should be... I will add.

@ZedThree could you push the changes you just made please?

Reply:
While looking at `Field3D::operator=(const Field2D &rhs)`, I have a question; should we set `fieldmesh = rhs.getMesh();`, or `ASSERT1(fieldmesh == rhs.getMesh());` in this operator?

Reply:
Done

Reply:
I think to be consistent with other places this should probably be an assert. If it is an assert we should ensure we assert for `Field3D::operator=(const Field3D &rhs)` etc. as well to be consistent.